mono/metadata/custom-attrs.c

```diff
@@ -767,14 +767,16 @@ decode_blob_value_checked (const char *ptr, const char *endp, guint32 *value_out
 	return is_ok (error);
 }
 
-static MonoObject*
-create_custom_attr (MonoImage *image, MonoMethod *method, const guchar *data, guint32 len, MonoError *error)
+static void
+create_custom_attr (MonoObjectHandle *result, MonoArrayHandle *result_array, int result_index, MonoImage *image, MonoMethod *method, const guchar *data, guint32 len, MonoError *error)
```

Comment:
1. I don't understand the "out" stuff. Just comments-via-naming, but always the same thing? ok.
2. I put the results first like assignment/memcpy, and this is replacing assignment, but ok.
3. ok. I think I didn't realize at first I'd have both.
more generally: I don't like extra single-use functions, and having to name them, but again this could be because I only later found the second user...and I have to check if it was also in a loop, and I never came up with good names for the loop helpers, so this avoids coming up with more similar names. I can try again here.

Reply:
btw 3 isn't right I think -- both can be non-null, but they don't tend to be. Anyway let me see if I can make it a little better.

Reply:
This should clear up a lot.
```
static void
create_custom_attr_into_array (MonoImage *image, MonoMethod *method, const guchar *data,
	guint32 len, MonoArrayHandle array, int index, MonoError *error)
{
	// This function serves to avoid creating handles in a loop.
	HANDLE_FUNCTION_ENTER ();
	MonoObjectHandle attr = create_custom_attr (image, method, data, len, error);
	MONO_HANDLE_ARRAY_SETREF (array, index, attr);
	HANDLE_FUNCTION_RETURN ();
}
```